spark3/src/main/java/org/apache/iceberg/spark/PathIdentifier.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import java.util.List;
+import org.apache.iceberg.relocated.com.google.common.base.Splitter;
+import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
+import org.apache.spark.sql.connector.catalog.Identifier;
+
+public class PathIdentifier implements Identifier {
+  private final String[] namespace;
+  private final String location;
+  private final String name;
+
+  public PathIdentifier(String location) {
+    this.location = location;
+    List<String> pathParts = Splitter.on("/").splitToList(location);
+    name = Iterables.getLast(pathParts);
+    namespace = pathParts.size() > 1 ? pathParts.subList(0, pathParts.size() - 1).toArray(new String[0]) :
+        new String[0];
```

Comment:
I think that the namespace should be the location, not an array of path parts. This would create really weird namespaces, like `["s3:", "", "bucket", "path", "to"]`.

Reply:
agreed that it shouldn't be `["s3:", "", "bucket", "path", "to"]` but I am not sure why it would be the full location. Then we have `namespace` returning `{location}` the new public method `location` returning `location` and the `name` method returning just the filename. To me `namespace` + `name` should equal location. I have updated this patch w/ my suggestion but happy to revert.

Reply:
Shouldn't namespace be just `iceberg` and name `location` to match built-in sources?

Reply:
I prefer @aokolnychyi's suggestion over splitting filename from directories. Then we dont have the extra `location` method on `PathIdentifier` too.

Reply:
> Shouldn't namespace be just iceberg and name location to match built-in sources?

I don't think so. Those identifiers would not be passed to Iceberg as `PathIdentifier` so there isn't a strong reason to make Iceberg's `PathIdentifier` use the same convention. They are all "iceberg" identifiers.

> To me namespace + name should equal location.

I'm fine either way. I do think that it makes sense to omit it from the namespace (["s3://bucket/path/to"] and "table"). The main requirement is that the location should be available unmodified when this is used.

Sounds like we agree that the table name should be the last directory name to make the [subquery alias](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L972) work.

Reply:
cool, have left it as is. Thanks for the pointer to subquery alias!